extensions/jdbc/jdbc-derby/jdbc-derby-runtime/src/main/java/io/quarkus/jdbc/derby/runtime/graal/ClientDriver_PlaceholderSubstitution.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.jdbc.derby.runtime.graal;
+
+import org.apache.derby.jdbc.ClientDriver;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(ClientDriver.class)
+final public class ClientDriver_PlaceholderSubstitution {
+
+    @Substitute
+    public boolean jdbcCompliant() {
+        // TODO: Just overriding this simple method for now so that we override at least one thing
+        // Need to check w/ Quarkus team on how to properly investigate what substitutions are necessary
+        return true;
```

Comment:
Well, if you don't need overrides, it's perfect and you just remove that class :).

Reply:
I am skeptical that 0 overrides are needed for Derby support, so I left this template in place for now. I *think* it all works in native mode (at least my golden-path JPA integration test), but I'd like someone to confirm that I'm actually getting lucky not needing any substitutions rather than doing something wrong on my end.

My understanding is that if substitutions were required for golden path usage, I would be seeing some sort of failure at deployment time or runtime when I run the jpa-derby integration test. Is this assumption correct?

Reply:
it's an interesting question.

We mostly rely on the graalvm native-image compiler to tell us: it's extremely strict on some things, so typically if it compiles it means the whole codebase is safe.

However there are exceptions to this; the most notable one is Reflection: it can't tell if any reflective code is safe, so it assumes it is - we'll get a runtime exception when we miss such cases.

So while having a deep familiarity with the library would help to give better reassurance, a simple integration test passing is typically going a far way - even if it's just a "golden path".

You could explore the codebase by grepping it for known nasty ones; that could lead to craft some tricky paths for which we might want some more integration tests, but we normally go to such lenghts only for libraries which are critical for production. For Derby and similar tools, we can also simply assume that it's fine until a bug is found.

Reply:
Also: with the other drivers I had issues with static initializers starting background threads and such, or acccess to security libraries not available.

GraalVM will check for such things at build, so if it builds fine we're good on such fronts.

Reply:
Yeah, I would remove that file. We have a ton of examples to look at if we need to add substitutions at some point.

Reply:
thanks for the explanation @Sanne, makes sense! I will go ahead and remove the file before I move this PR out of draft mode